company.py

```diff
@@ -0,0 +1,182 @@
+            {"first_name": "Michelle", "last_name": "Frey", "position": "Middle HR", "salary_rub": 75000},
+            {"first_name": "Kevin", "last_name": "Jimenez", "position": "Middle HR", "salary_rub": 70000},
+            {"first_name": "Nicole", "last_name": "Riley", "position": "HRD", "salary_rub": 120000},
+        ]
+    },
+    {
+        "title": "IT department",
+        "employers": [
+            {"first_name": "Christina", "last_name": "Walker", "position": "Python dev", "salary_rub": 80000},
+            {"first_name": "Michelle", "last_name": "Gilbert", "position": "JS dev", "salary_rub": 85000},
+            {"first_name": "Caitlin", "last_name": "Bradley", "position": "Teamlead", "salary_rub": 950000},
+            {"first_name": "Brian", "last_name": "Hartman", "position": "CTO", "salary_rub": 130000},
+        ]
+    },
+]
+
+taxes = [
+    {"department": None, "name": "vat", "value_percents": 13},
+    {"department": "IT Department", "name": "hiring", "value_percents": 6},
+    {"department": "BizDev Department", "name": "sales", "value_percents": 20},
+]
+
+
+def last_name_vowel(departments):
+    last_name_vowel_final = set()
+    for department in departments:
+        for employers_info in department['employers']:
+            if employers_info['last_name'][-1] in 'a, e, i, o, u, y':
+                last_name_vowel_final.add(employers_info['first_name'])
+    return last_name_vowel_final
+
+def salary_women(departments):
+    salary_women_dep = []
+    for departmen in departments:
+        dep = ''
+        count_women = 0
+        salary_women = 0
+        for employers_info in departmen['employers']:
+            if employers_info['first_name'] in "Michelle, Nicole":
+                count_women += 1
+                salary_women += employers_info['salary_rub']
+                dep = departmen['title']
+        salary_women_dep.append((dep, int(salary_women/count_women) ))
+    return salary_women_dep
+
+
+def position_ninety(departments):
+    list_pos_ninety = set()
+    for employers in departments:
+        for employers_info in employers['employers']:
+            if employers_info['salary_rub'] > 90000:
+                list_pos_ninety.add(employers_info['position'])
+    return list_pos_ninety
+
+def aver_salary_company(departments):
+    money_max = 0
+    count_employers = 0
+    for employers in departments:
+        for employers_info in employers['employers']:
+            money_max += employers_info['salary_rub']
+            count_employers += 1
+    return int(money_max/count_employers)
+
+def min_aver_max_salary(departments):
+    dep_money = []
+    for employers in departments:
+        title = employers['title']
+        money_min = 0
+        money_all = 0
+        money_max = 0
+        count_employers = 0
+        for employers_info in employers['employers']:
+            if money_min == 0 or money_min > employers_info['salary_rub']:
```

Comment:
чтобы 5 раз не писать employers_info['salary_rub'], я бы её в начале цикла записал в переменную
```suggestion
            employer_salary = employers_info['salary_rub']
            if money_min == 0 or money_min > employer_salary:
```
это удобно ещё потому, что если в словаре ключ зарплаты поменяется и станет например 'salary' вместо 'salary_rub', достаточно будет его поменять в одном месте, а не в пяти.